File: Data Structures & Algorithms/two-integer-sum/submission-0.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> storage = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++){
            int diff = target - nums[i];
            if(storage.ContainsKey(diff)) return new int[] {storage[diff], i};
            storage[nums[i]] = i;
        }
        return new int[]{};
    }
}

File: Data Structures & Algorithms/two-integer-sum-ii/submission-0.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        Map<Integer, Integer> map = new HashMap<Integer, Integer>();
        for(int i = 0; i < numbers.length; i++){
            int diff = target - numbers[i];
            if(map.containsKey(diff)) return new int[]{ map.get(diff), i + 1};
            else map.put(numbers[i], i + 1);
        }
        return new int[0];
    }
}
